README.md

```diff
@@ -235,6 +235,12 @@ brew install keith/formulae/git-pile
 
 [template]: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository
 
+### GitLab support
+
+- You can use `git-pile` with GitLab. Enable GitLab mode by running
+ `git config pile.gitlabModeEnabled true`. Also set your GitLab URL via
+ `git config pile.gitlabMergeRequestsURL "https://gitlab.com/{group}/{project}/merge_requests/"`
```

Comment:
is this derivable from the remote at all? or is that not a 1:1 mapping that would be a good default?

Reply:
We can definitely try to derive it by default! I'm not sure that we can cover all cases automatically (so ability to override URL via config can be useful), but origin can look like:

```
https://gitlab.com/companyname/ios.git                    // HTTPS origin
git@gitlab.com:companyname/ios.git                        // SSH origin

https://gitlab.com/companyname/ios/-/merge_requests/new    // Desired result
```

Or for on premise GitLab on custom domain:
```
https://gitlab.companydomain.app/mobile/ios.git                     // HTTPS origin
git@git.gitlab.companydomain.app:mobile/ios.git                     // SSH origin

https://gitlab.companydomain.app/mobile/ios/-/merge_requests/new    // Desired result
```

One more example:
```
https://gitlab.com/companyname/sre/protobuf.git                     // HTTPS origin
git@gitlab.com:companyname/sre/protobuf.git                         // SSH origin

https://gitlab.com/companyname/sre/protobuf/-/merge_requests/new    // Desired result
```

I think it's better to make it in separate PR.

Reply:
separate PR sounds good. i guess the biggest reason for allowing overriding is if the url doesn't contain `gitlab` or `github` at all? because otherwise these look pretty derivable

Reply:
Something like https://github.com/MontakOleg/git-pile/pull/2, but patterns could be more strict.